### PR TITLE
Fix issue with titlebar sample setting caption button background to the wrong color

### LINKED ISSUE #1355: Title Bar custom background & foreground colors not properly working
### Which version of the app?

WinUI 3 Gallery

### Description

- When switching the background color of the title bar, it doesn't evenly fill up the entire title bar and gets cut off at the title bar logo
- The foreground color controls do not work at all and have no functionality with the new top-level UI Element
- Resetting to the system title bar causes the window minimize, restore/maximize, and close button controls to become impossible to see

A video and screenshot have been attached showing the following issues.

### Screenshots

SystemTitleBar_Issue_9-9-23

https://github.com/microsoft/WinUI-Gallery/assets/54720001/a6a4a250-95db-42c4-8183-eb4f903fd8b9

### Windows version

Insider Build (xxxxx)

### Additional context

These issues were tested and found on 23H2 (22631.2271) in the video, but I also found them on my laptop running 22H2 release. These issues were found in WinUI3 gallery version 2.2.0.0

### PR DESCRIPTION
**Pull request by chingucoding**
## Description

Closes #1355 
## Motivation and Context

This fixes the issues mentioned in #1355 and also the fact that we showed two titlebars in some cases.
## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### CODE REVIEW

**Comment by pratikone** [comment]
Thanks for the PR. Could you verify if this issue still exists with newer builds of WInUI gallery  (built from repo). With merger with appwindow titlebar, I think this won't be an issue. But I could be wrong so it is worth checking it out.

**Comment by chingucoding** [comment]
@pratikone, I'm afraid this PR is still relevant, the issues were still present when I tested.

File: WinUIGallery/ControlPages/TitleBarPage.xaml.cs [hunk 1]
```diff
@@ -146,9 +150,28 @@ public void UpdateTitleBarColor()
...
+            if(currentBgColor == Colors.Transparent)
+            {
+                // If the current background is null, we want to revert to the default titlebar which is achieved using null as color
+                TitleBarHelper.SetBackgroundColor(window, null);
+            }
+            else
+            {
+                TitleBarHelper.SetBackgroundColor(window, currentBgColor);
+            }
+            TitleBarHelper.SetForegroundColor(window, currentFgColor);
```
Comment (karkarl) [inline WinUIGallery/ControlPages/TitleBarPage.xaml.cs:174]: >SetForegroundColor [](http://example.com/codeflow?start=27&length=18)

nit: Capitalize first letter and add period to comment.

File: WinUIGallery/ControlPages/TitleBarPage.xaml.cs [hunk 2]
```diff
@@ -146,9 +150,28 @@ public void UpdateTitleBarColor()
...
             (titleBarElement as Border).Background = new SolidColorBrush(currentBgColor); // changing titlebar uielement's color
+            if(currentFgColor != Colors.Transparent)
+            {
+                (titleBarAppNameElement as TextBlock).Foreground = new SolidColorBrush(currentFgColor);
+            }
+            else
+            {
+                (titleBarAppNameElement as TextBlock).Foreground = Application.Current.Resources["TextFillColorPrimaryBrush"] as SolidColorBrush;
+            }
             TitleBarHelper.SetCaptionButtonColors(window, currentFgColor);
```
Comment (karkarl) [inline WinUIGallery/ControlPages/TitleBarPage.xaml.cs:164]: nit: Add extra line before and after if / else block for readability.

File: WinUIGallery/ControlPages/TitleBarPage.xaml.cs (hunk 1 above)
Comment (chingucoding) [reply WinUIGallery/ControlPages/TitleBarPage.xaml.cs:178]: I'm afraid I'm not sure which first letter to capitalize here 😅

**Review by pratikone — APPROVED** [review]
looks good. thanks @karkarl  for verifying the issue too

**Comment by chingucoding** [comment]
Thank you @pratikone for the CI rerun and your feedback on this PR!

**Pull request merged**